spec/requests/api/graphql_spec.rb

```diff
@@ -0,0 +1,664 @@
+require "manageiq/api/common/graphql"
+
+RSpec.describe ManageIQ::API::Common::GraphQL, :type => :request do
+  let!(:graphql_endpoint) { "/api/v1.0/graphql" }
+
+  let!(:ext_tenant)   { rand(1000).to_s }
+  let!(:tenant)       { Tenant.create!(:name => "tenant_a", :external_tenant => ext_tenant) }
+  let!(:identity)     { Base64.encode64({'identity' => { 'account_number' => ext_tenant }}.to_json) }
+  let!(:headers)      { { "CONTENT_TYPE" => "application/json", "x-rh-identity" => identity } }
+
+  let!(:source_typeR) { SourceType.create(:name => "rhev", :product_name => "RedHat Virtualization", :vendor => "redhat") }
+  let!(:source_typeV) { SourceType.create(:name => "vmware", :product_name => "VmWare vCenter", :vendor => "vmware") }
+
+  let!(:source_a1)    { Source.create!(:tenant => tenant, :uid => "1", :name => "source_a1", :source_type => source_typeR) }
+  let!(:source_a2)    { Source.create!(:tenant => tenant, :uid => "2", :name => "source_a2", :source_type => source_typeR) }
+  let!(:source_b1)    { Source.create!(:tenant => tenant, :uid => "3", :name => "source_b1", :source_type => source_typeR) }
+  let!(:source_b2)    { Source.create!(:tenant => tenant, :uid => "4", :name => "source_b2", :source_type => source_typeR) }
+  let!(:source_b3)    { Source.create!(:tenant => tenant, :uid => "5", :name => "source_b3", :source_type => source_typeR) }
+
+  let!(:endpoint_a21) { Endpoint.create!(:tenant => tenant, :source => source_a2, :host => "www.source_a2.com", :port => "121", :role => "web_lb1") }
```

Comment:
Minor, but for the URLs, prefer subdomains of example.com (This way we don't accidentally point to "real" domains), so in this case `source_a2.example.com`